lib/blogs_api/user/create.ex

```diff
@@ -0,0 +1,33 @@
+defmodule BlogsApi.User.Create do
+  @moduledoc """
+  Criação do Módulo bem encapsulado e específico para a ação do Create
+  """
+  alias Elixir.BlogsApi.{Repo, User}
+
+  def call(params) do
+    params
+    |> User.changeset()
+    |> Repo.insert()
+    |> email_exists?()
+  end
```

Comment:
Aqui você está inserindo o usuário no banco antes de garantir que o email é único? Não seria o contrário?
```suggestion
  def call(params) do
    params
    |> User.changeset()
    |> email_exists?()
    |> Repo.insert()
  end
```